java/src/main/java/com/powsybl/dataframe/network/adders/NetworkUtils.java

```diff
@@ -28,6 +26,28 @@ public static VoltageLevel getVoltageLevelOrThrow(Network network, String id) {
         return voltageLevel;
     }
 
+    public static VoltageLevel getVoltageLevelOrThrowWithBusOrBusbarSectionId(Network network, int row, StringSeries voltageLevels, StringSeries busOrBusbarSections) {
+        if (voltageLevels == null) {
+            if (busOrBusbarSections.get(row) != null) {
+                Identifiable<?> busOrBusbarSection = network.getIdentifiable(busOrBusbarSections.get(row));
+                if (busOrBusbarSection == null) {
+                    throw new PowsyblException(String.format("Identifiable %s not found.", busOrBusbarSections.get(row)));
+                }
+                if (busOrBusbarSection instanceof BusbarSection) {
+                    return ((BusbarSection) busOrBusbarSection).getTerminal().getVoltageLevel();
+                } else if (busOrBusbarSection instanceof Bus) {
+                    return ((Bus) busOrBusbarSection).getVoltageLevel();
+                } else {
+                    throw new PowsyblException(String.format("Unsupported type %s for identifiable %s", busOrBusbarSection.getType(), busOrBusbarSection.getId()));
+                }
+            } else {
+                throw new PowsyblException("voltage_level_id is missing");
```

Comment:
should it be more explicite like voltage level or bus id or busbar section id is missing ?